pkg/util/podstore/podstore_test.go

```diff
@@ -0,0 +1,549 @@
+		fakeClock.SetTime(expectedDeleteTime.Add(10 * time.Millisecond))
+		assert.Eventuallyf(t, func() bool {
+			return podStore.podsToDelete.Len() == 1
+		}, 1*time.Second, 10*time.Millisecond, "Pod is not added to PodsToDelete")
+	})
+}
+
+func Test_checkDeletedPod(t *testing.T) {
+	tests := []struct {
+		name           string
+		obj            interface{}
+		expectedResult *v1.Pod
+		expectedErr    string
+	}{
+		{
+			name:        "object is not DeletedFinalStateUnknown",
+			obj:         node,
+			expectedErr: "received unexpected object: ",
+		},
+		{
+			name:        "object in DeletedFinalStateUnknown is not Pod",
+			obj:         cache.DeletedFinalStateUnknown{Obj: node},
+			expectedErr: "DeletedFinalStateUnknown object is not of type Pod",
+		},
+		{
+			name:           "valid case",
+			obj:            cache.DeletedFinalStateUnknown{Obj: pod1},
+			expectedResult: pod1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			pod, err := (&PodStore{}).checkDeletedPod(tt.obj)
+			if tt.expectedErr != "" {
+				assert.ErrorContains(t, err, tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+				assert.Equal(t, tt.expectedResult, pod)
+			}
+		})
+	}
+}
+
+func Test_GetPodByIPAndTime(t *testing.T) {
+	k8sClient := fake.NewSimpleClientset()
+	podInformer := coreinformers.NewPodInformer(k8sClient, metav1.NamespaceAll, 0, cache.Indexers{})
+	tests := []struct {
+		name           string
+		ip             string
+		startTime      time.Time
+		expectedResult *v1.Pod
+	}{
+		{
+			name:           "no Pod in the Pod store",
+			ip:             "1.3.5.7",
+			startTime:      refTime,
+			expectedResult: nil,
+		},
+		{
+			name:           "find only one Pod in the Pod store - correct startTime",
+			ip:             "5.6.7.8",
+			startTime:      refTime.Add(-time.Minute),
+			expectedResult: pod2,
+		},
+		{
+			name:           "find only one Pod in the Pod store - incorrect startTime",
+			ip:             "5.6.7.8",
+			startTime:      refTime.Add(time.Minute),
+			expectedResult: pod2,
+		},
+		{
+			name:           "find current Pod in the Pod store",
+			ip:             "1.2.3.4",
+			startTime:      refTime.Add(time.Minute),
+			expectedResult: pod1,
+		},
+		{
+			name:           "find previous Pod in the Pod store",
+			ip:             "1.2.3.4",
+			startTime:      refTime.Add(-time.Minute),
+			expectedResult: pod4,
+		},
+		{
+			name:           "cannot find the Pod in the Pod store - SearchTime < CreationTime",
+			ip:             "1.2.3.4",
+			startTime:      refTime.Add(-time.Minute * 10),
+			expectedResult: nil,
+		},
+		{
+			name:           "cannot find the Pod in the Pod store - SearchTime > DeletionTime",
+			ip:             "1.2.3.4",
+			startTime:      refTime,
+			expectedResult: nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			podStore := NewPodStore(podInformer)
+			err := podStore.pods.Add(pod1)
+			require.NoError(t, err)
```

Comment:
use `require.NoError(t, podStore.pods.Add(pod1))` for less verbosity
same below

Reply:
Updated. Thanks.